File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        
        prev=dummy=ListNode()

        while list1!=None and list2!=None:
            if list1.val<=list2.val:
                prev.next=list1
                list1=list1.next
                prev=prev.next
            else:
                prev.next=list2
                list2=list2.next
                prev=prev.next

        if list1!=None:
            prev.next=list1
        else:
            prev.next=list2

        return dummy.next

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2) {
    
    struct ListNode dummy;

    struct ListNode* prev=&dummy;

    while(l1!=NULL && l2!=NULL){

        if(l1->val<=l2->val){
            prev->next=l1;
            l1=l1->next;
            prev=prev->next;
        }
        else{
            prev->next=l2;
            l2=l2->next;
            prev=prev->next;
        }
    }

    if(l1!=NULL){
        prev->next=l1;
    }
    else{
        prev->next=l2;
    }

    return dummy.next;

}